### PR TITLE
Moves some stuff that belongs in dbcore to dbcore

### PR DESCRIPTION
**Pull request by Cyberboss**

### CODE REVIEW

File: code/orphaned_procs/dbcore.dm [hunk 1]
```diff
@@ -55,15 +56,35 @@ DBConnection/New(dbi_handler,username,password_handler,cursor_handler)
 	src.default_cursor = cursor_handler
 	_db_con = _dm_db_new_con()
 
-DBConnection/proc/Connect(dbi_handler=src.dbi,user_handler=src.user,password_handler=src.password,cursor_handler)
+DBConnection/proc/Connect()
+	if(IsConnected())
+		return TRUE
+
+	if(failed_connections > FAILED_DB_CONNECTION_CUTOFF)	//If it failed to establish a connection more than 5 times in a row, don't bother attempting to connect anymore.
```
Comment (Jordie0608) [inline code/orphaned_procs/dbcore.dm]: You never actually increment `failed_connections` anymore.

**Review by Jordie0608 — CHANGES_REQUESTED** [review]

File: code/orphaned_procs/dbcore.dm (hunk 1 above)
Comment (Cyberboss) [inline code/orphaned_procs/dbcore.dm]: thank for reminding me

**Review by Jordie0608 — APPROVED** [review]

File: code/controllers/subsystem/garbage.dm [hunk 2]
```diff
@@ -350,7 +350,7 @@ var/datum/subsystem/garbage_collector/SSgarbage
 //if find_references isn't working for some datum
 //update this list using tools/DMTreeToGlobalsList
 /datum/proc/find_references_in_globals()
-	SearchVar(failed_db_connections)
+	SearchVar(last_irc_status)
```
Comment (optimumtact) [inline code/controllers/subsystem/garbage.dm]: hmm this was in the master for the conflicts

**Comment by optimumtact** [comment]
this conflict editor on github is useless

**Pull request merged**